### PR TITLE
Add json_parse, json_format; remove cast between json and varchar

### PR DESCRIPTION
**Pull request by haozhun**
Issue #3197

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonFunctions.java [hunk 1]
```diff
@@ -60,13 +70,28 @@ public static JsonPath castToJsonPath(@SqlType(StandardTypes.VARCHAR) Slice patt
...
     @SqlType(StandardTypes.VARCHAR)
-    public static Slice castToVarchar(@SqlType(StandardTypes.JSON) Slice slice)
+    public static Slice fromJson(@SqlType(StandardTypes.JSON) Slice slice)
     {
         return slice;
     }
 
+    @ScalarFunction
+    @SqlType(StandardTypes.JSON)
+    public static Slice toJson(@SqlType(StandardTypes.VARCHAR) Slice slice) throws IOException
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonFunctions.java]: put `throws` on the next line

**Comment by cberner** [comment]
So, are we not going to have a `CAST` between json and varchar, or are we going to add it back later?

**Comment by electrum** [comment]
I'm not sure `from_json` is the right name. How about `to_char`?

**Comment by electrum** [comment]
Or maybe `json_format`?

**Comment by electrum** [comment]
We could have a flag to pretty print the JSON.

**Comment by haozhun** [comment]
I agree that `json_parse` and `json_format` are better names. We can go with this if nothing better comes up. It's not a perfect name though because `json_format` feels counter-intuitive when one is trying to find the function that will turn his JSON document to VARCHAR. In addition, It does feel that `parse` and `format` is better than `from` and `to` because the two functions don't round trip.

**Comment by haozhun** [comment]
Rebased and renamed

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java [hunk 2]
```diff
@@ -571,11 +572,13 @@ protected Type visitGenericLiteral(GenericLiteral node, AnalysisContext context)
                 throw new SemanticException(TYPE_MISMATCH, node, "Unknown type: " + node.getType());
             }
 
-            try {
-                functionRegistry.getCoercion(VARCHAR, type);
-            }
-            catch (IllegalArgumentException e) {
-                throw new SemanticException(TYPE_MISMATCH, node, "No literal form for type %s", type);
+            if (type != JSON) {
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java]: you need to use `.equals()` for types. Also, does the else branch need to provide the `json_parse` function?

File: presto-main/src/main/java/com/facebook/presto/sql/planner/LiteralInterpreter.java [hunk 3]
```diff
@@ -215,6 +216,16 @@ protected Object visitGenericLiteral(GenericLiteral node, ConnectorSession sessi
                 throw new SemanticException(TYPE_MISMATCH, node, "Unknown type: " + node.getType());
             }
 
+            if (type == JSON) {
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/LiteralInterpreter.java]: use `.equals()` not `==`

**Comment by cberner** [comment]
Had a couple comments and questions

**Comment by haozhun** [comment]
Resolved comments and questions

File: presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java [hunk 4]
```diff
@@ -187,6 +188,13 @@ protected RowExpression visitGenericLiteral(GenericLiteral node, Void context)
                 throw new IllegalArgumentException("Unsupported type: " + node.getType());
             }
 
+            if (type == JSON) {
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java]: use `.equals()`

**Comment by cberner** [comment]
Had one comment. Also the tests are failing. Otherwise, looks good

**Pull request merged**